packages/core/admin/admin/src/content-manager/history/components/VersionDetails.tsx

```diff
@@ -17,12 +20,35 @@ interface VersionHeaderProps {
 }
 
 const VersionHeader = ({ headerId, version }: VersionHeaderProps) => {
-  const { formatMessage } = useIntl();
+  const { formatMessage, formatDate, formatTime } = useIntl();
+  const { slug } = useParams<{
+    slug: UID.ContentType;
+  }>();
+  const { layout } = useContentTypeLayout(slug);
+
+  // The loading state is handled by the parent component
+  if (!layout) return null;
+
+  const mainFieldValue = version.data[layout.contentType.settings.mainField || 'id'] as string;
+  const formattedSubtitle = formatMessage(
+    {
+      id: 'content-manager.history.version.subtitle',
+      defaultMessage: '{hasLocale, select, true {{subtitle}, in {locale}} other {{subtitle}}}',
+    },
+    {
+      hasLocale: Boolean(version.locale),
+      subtitle: `${mainFieldValue} (${layout.contentType.info.singularName})`,
+      locale: version.locale?.name,
+    }
+  );
 
   return (
     <HeaderLayout
       id={headerId}
-      title={`History version ${version.id}`}
+      title={`${formatDate(new Date(version.createdAt))}, ${formatTime(
+        new Date(version.createdAt)
+      )}`}
```

Comment:
I'm not sure you need to do this concatenation yourself, I think you could use formatDate only and give hour and minute options, or give it a timeStyle